pydrawise/client.py

```diff
@@ -420,36 +441,47 @@ async def get_water_use_summary(
-                and entry.run_event.zone is not None
-                and entry.run_event.reported_water_usage is not None
-            ):
-                active_use = entry.run_event.reported_water_usage.value
-                if summary.unit == "":
-                    summary.unit = entry.run_event.reported_water_usage.unit
-                summary.total_active_use += active_use
-                summary.active_use_by_zone_id.setdefault(entry.run_event.zone.id, 0)
-                summary.active_use_by_zone_id[entry.run_event.zone.id] += active_use
+            if entry.run_event is not None and entry.run_event.zone is not None:
+                if (
+                    entry.run_event.reported_water_usage is not None
+                    and has_flow_sensors
+                ):
+                    active_use = entry.run_event.reported_water_usage.value
+                    if summary.unit == "":
+                        summary.unit = entry.run_event.reported_water_usage.unit
+                    summary.total_active_use += active_use
+                    summary.active_use_by_zone_id.setdefault(entry.run_event.zone.id, 0)
+                    summary.active_use_by_zone_id[entry.run_event.zone.id] += active_use
+
+                active_time = entry.run_event.reported_duration
+                summary.total_active_time += active_time
+                summary.active_time_by_zone_id.setdefault(
+                    entry.run_event.zone.id, timedelta()
+                )
+                summary.active_time_by_zone_id[entry.run_event.zone.id] += active_time
 
         # total active and inactive water use
-        for sensor in result["controller"]["sensors"]:
-            if (
-                "FLOW" in sensor["model"]["sensorType"]
-                and "flowSummary" in sensor
-                and (flow_summary := sensor["flowSummary"]) is not None
-            ):
-                summary.total_use += flow_summary["totalWaterVolume"]["value"]
-                if summary.unit == "":
-                    summary.unit = flow_summary["totalWaterVolume"]["unit"]
-
-        # Correct for inaccuracies. The watering report and flow summaries are not always
-        # updated with the same frequency.
-        if summary.total_use > summary.total_active_use:
-            summary.total_inactive_use = summary.total_use - summary.total_active_use
-        else:
-            summary.total_use = summary.total_active_use
+        if has_flow_sensors:
+            for sensor in result["controller"]["sensors"]:
+                if (
+                    "FLOW" in sensor["model"]["sensorType"]
+                    and "flowSummary" in sensor
+                    and (flow_summary := sensor["flowSummary"]) is not None
+                ):
+                    summary.total_use += flow_summary["totalWaterVolume"]["value"]
+                    if summary.unit == "":
+                        summary.unit = flow_summary["totalWaterVolume"]["unit"]
```

Comment:
How about we add a new type to schema.py:

```py
@dataclass
@type_name("Sensor")
class SensorWithFlowSummary(Sensor):
    """A Sensor, as returned by its `flowSummary` method."""

    flow_summary: Optional[SensorFlowSummary] = _optional_field(
        default_factory=SensorFlowSummary
    )
```

Then this can change to:

```suggestion
            for sensor_json in result["controller"]["sensors"]:
                sensor = deserialize(SensorWithFlowSummary, sensor_json)
                if sensor.flow_summary and sensor.model.sensor_type == CustomSensorTypeEnum.FLOW:
                    summary.total_use += sensor.flow_summary.total_water_volume.value
                    if summary.unit == "":
                        summary.unit = sensor.flow_summary.total_water_volume.unit
```

Reply:
Fixed.